astropy/coordinates/tests/test_angles.py

```diff
@@ -1092,3 +1092,48 @@ def test_str_repr_angles_nan(cls, input, expstr, exprepr):
     # Deleting whitespaces since repr appears to be adding them for some values
     # making the test fail.
     assert repr(q).replace(" ", "") == f'<{cls.__name__}{exprepr}>'.replace(" ","")
+
+
+@pytest.mark.parametrize("sign", (-1, 1))
+@pytest.mark.parametrize(
+    "value,expected_value,dtype,expected_dtype",
+    [
+        (np.pi / 2, np.pi / 2, None, np.float64),
+        (np.pi / 2, np.pi / 2, np.float64, np.float64),
+        (np.float32(np.pi / 2), np.float32(np.pi / 2), None, np.float32),
+        (np.float32(np.pi / 2), np.float32(np.pi / 2), np.float32, np.float32),
+        # these cases would require coercing the float32 value to the float64 value
+        # making validate have side effects, so it's not implemented for now
+        # (np.float32(np.pi / 2), np.pi / 2, np.float64, np.float64),
+        # (np.float32(-np.pi / 2), -np.pi / 2, np.float64, np.float64),
+    ]
+)
+def test_latitude_limits(value, expected_value, dtype, expected_dtype, sign):
+    """
+    Test cases for issue #13708
+    """
+    # this prevents upcasting to float64 as sign * value would do
+    if sign < 0:
+        value = -value
+        expected_value = -expected_value
+
+    result = Latitude(value, u.rad, dtype=dtype)
+    assert result.value == expected_value
+    assert result.dtype == expected_dtype
+    assert result.unit == u.rad
+
+
+@pytest.mark.parametrize(
+    "value,dtype",
+    [
+        (0.50001 * np.pi, np.float32),
+        (np.float32(0.50001 * np.pi), np.float32),
+        (0.50001 * np.pi, np.float64),
+    ]
+)
+def test_latitude_out_of_limits(value, dtype):
+    """
+    Test cases for issue #13708
+    """
+    with pytest.raises(ValueError):
```

Comment:
It would be good to test the error message here, and the expected error message would also describe what this test is meant for so that the docstring would not be required.

Reply:
done